desktop/apps/consignments2/client/reducers.js

```diff
@@ -47,6 +63,14 @@ function submissionFlow (state = initialState, action) {
         return state
       }
     }
+    case actions.UPDATE_INPUTS: {
+      const existingInputs = clone(state.inputs)
+      const newInputs = merge(existingInputs, action.payload.inputs)
+
+      return u({
+        inputs: newInputs
+      }, state)
```

Comment:
Shouldn't need to clone or merge; can simply do 
```javascript
return({
  ...state.inputs,
  ...action.payload.inputs
}, state)
```
the `...` (spread) operator creates a shallow clone, and using two together desugars into `Object.assign({}, a, b)` which is ES6's `merge` function.